governance/proposals/010-arbitrum-l1-l2-messaging.js

```diff
@@ -0,0 +1,263 @@
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'payable',
+    type: 'function',
+  },
+]
+
+/**
+ * Set up: instantiate L1 / L2 wallets connected to providers
+ */
+const walletPrivateKey = process.env.DEVNET_PRIVKEY
+const l1Provider = new ethers.JsonRpcProvider(process.env.L1RPC)
+const l2Provider = new ethers.JsonRpcProvider(process.env.L2RPC)
+// const l1Wallet = new ethers.Wallet(walletPrivateKey, l1Provider)
+const l2Wallet = new ethers.Wallet(walletPrivateKey, l2Provider)
+
+module.exports = async () => {
+  await arbLog('Cross-chain Proposer')
+  addDefaultLocalNetwork()
+
+  const l2Network = await getL2Network(l2Provider)
+  const ethBridger = new EthBridger(l2Network)
+  const inboxAddress = ethBridger.l2Network.ethBridge.inbox
+  const ARBTokenAddressOnL2 = '0x912CE59144191C1204E64559FE8253a0e49E6548' // ARB TOKEN ADDRESS ON ARBITRUM ONE
+  const grantsContractAddress = '0x00D5E0d31d37cc13C645D86410aB4cB7Cb428ccA'
+  const L2Alias = '0x28ffDfB0A6e6E06E95B3A1f928Dc4024240bD87c' // Timelock Alias Address on L2
+  const L1TimelockContract = '0x17EEDFb0a6E6e06E95B3A1F928dc4024240BC76B' // Timelock Address mainnet
+
+  const L2TokenContract = new ethers.Contract(
+    ARBTokenAddressOnL2,
+    ERC20_ABI,
+    l2Wallet
+  ).connect(l2Wallet)
+
+  const balanceOf = await L2TokenContract.balanceOf(L2Alias)
+  console.log('ARB BALANCE::', ethers.formatEther(await balanceOf))
+
+  const tokenAmount = ethers.parseEther('1')
+
+  // Create an instance of the Interface from the ABIs
+  const iface_erc20 = new ethers.Interface(ERC20_ABI)
+  const iface_inbox = new ethers.Interface(INBOX_ABI)
+
+  // Encode the ERC20 Token transfer calldata
+  const transfer_calldata = iface_erc20.encodeFunctionData('transfer', [
+    grantsContractAddress,
+    tokenAmount,
+  ])
+  /**
+   * Now we can query the required gas params using the estimateAll method in Arbitrum SDK
+   */
+  const l1ToL2MessageGasEstimate = new L1ToL2MessageGasEstimator(l2Provider)
+
+  /**
+   * The estimateAll method gives us the following values for sending an L1->L2 message
+   * (1) maxSubmissionCost: The maximum cost to be paid for submitting the transaction
+   * (2) gasLimit: The L2 gas limit
+   * (3) deposit: The total amount to deposit on L1 to cover L2 gas and L2 call value
+   */
+  const L1ToL2MessageGasParams = await l1ToL2MessageGasEstimate.estimateAll(
+    {
+      from: L1TimelockContract,
+      to: ARBTokenAddressOnL2,
+      l2CallValue: 0,
+      excessFeeRefundAddress: l2Wallet.address,
+      callValueRefundAddress: l2Wallet.address,
+      data: transfer_calldata,
+    },
+    await getBaseFee(l1Provider),
+    l1Provider
+  )
+  console.log(':::::::::L1ToL2MessageGasParams::::::::::')
+  console.log(
+    'GasParams::::gasLimit',
+    L1ToL2MessageGasParams.gasLimit.toNumber()
+  )
+  console.log(
+    'GasParams::::maxSubmissionCost',
+    L1ToL2MessageGasParams.maxSubmissionCost.toNumber()
+  )
+  console.log(
```

Comment:
Could you please remove the console logs from the file ? Better to keep things clean

Reply:
All done